### PR TITLE
Hajj check

### PR DESCRIPTION
**Pull request by samdanikouser**
## Is this a Feature, Chore or Bug?
- Story

## Clearly and concisely describe the feature, chore or bug.
          [] I want to allocate Hajj leave only to Muslim staff members who have not performed Hajj before so that unqualified employees are not allocated Hajj leave
        
        Acceptance Criteria
        Given that a leave allocation is being created or edited and the leave type is "Hajj Leave"
        when the user clicks save
        then the system validates that the Employee is a muslim who has not performed Hajj. If these 2 conditions are not met, a message is shown and the process is aborted
        
        Notes:
        -Hajj Leave allocation is only allowed for a muslim staff Who has not performed Hajj before.
        -[Religion] and [Went to Hajj] are fields in the Employee doctype
        
        Consider moving [Went to Hajj] to "Personal" tab, after Religion and it should only be editable if [Religion] is set as "Muslim" (Depends on: eval: Religion == "Muslim").

## Analysis and design (optional)
[Analyse and attach the design documentation
](https://lucid.app/lucidchart/37a0e91b-1f10-40d0-bb6c-e7126f57eb57/edit?page=2tkbzzr9QS2v#)

## Solution description
---  Code changes in employee.js file of docstype.js ,added a check to religion if muslim to enable the went to hajj checkbox else to disable it .
--- Code changes in leave_allocation.py file for checking if religion is muslim to allocate the leave for that specific employee

## Is there a business logic within a doctype?
    - [] No

## Output screenshots (optional)
Screenshot 2024-10-14 at 11 21 27 AM

Screenshot 2024-10-14 at 11 21 39 AM
## Leave Allocation screenshort
Screenshot 2024-10-14 at 11 22 34 AM

##List out the areas affected by your code changes.
  -- Employee.js and leave_allocation.py file

## Is there any existing behavior change of other features due to this code change?
 Yes - The went to hajj field is been shifted to the personal details from the attedance in the employee profile page

## Did you delete custom field?
    - [] No

## Is patch required?
- [] No

## Which browser(s) did you use for testing?
  - [] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/overrides/leave_allocation.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        self.validate_employee_gender()
+        self.validate_hajj_leave()
+    
+    def validate_employee_gender(self):
+        gender  = frappe.db.get_value("Employee", self.employee, "gender","went_to_hajj")
+        if gender == "Male" and self.leave_type == "Maternity Leave":
+            frappe.throw("Maternity Leave allocation is only allowed for female workers.")
+
+    def validate_hajj_leave(self):
+        religion, went_to_hajj = frappe.db.get_value("Employee", self.employee, ["one_fm_religion", "went_to_hajj"])
```
Comment (NoBoneZ) [inline one_fm/overrides/leave_allocation.py]: Kindly avoid explosion, it caused issues in previous development iterations, you could use the , as_dict kwarg, then use the dictionary method, to get th eneeded fields

File: one_fm/overrides/leave_allocation.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+from hrms.hr.doctype.leave_allocation.leave_allocation import LeaveAllocation
+
+
+class LeaveAllocationOverride(LeaveAllocation):
+    def validate(self):
+        super().validate()
+        self.validate_employee_gender()
+        self.validate_hajj_leave()
+    
+    def validate_employee_gender(self):
```
Comment (NoBoneZ) [inline one_fm/overrides/leave_allocation.py]: kindly merge staging into this branch

File: one_fm/fixtures/doctype_layout.json [hunk 3]
```diff
@@ -1013,6 +1006,13 @@
     "parentfield": "fields",
     "parenttype": "DocType Layout"
    },
+   {
```
Comment (NoBoneZ) [inline one_fm/fixtures/doctype_layout.json:1009]: KIndly run migrations on your local bench, and confirm that this works as excpected

**Review by NoBoneZ — CHANGES_REQUESTED** [review]
KIndly check the comments

**Comment by Talleyrand333** [comment]
@samdanikouser Please check the file changes. there should not be so many changes for this.

**Pull request closed without merging**